lib/features/auth/presentation/widgets/security_settings.dart

```diff
@@ -196,6 +198,49 @@ class SecuritySettings extends StatelessWidget {
                             }
                           },
                         ),
+                        SwitchListTile(
+                          inactiveTrackColor: inactiveTrackColor,
+                          activeColor: activeColor,
+                          contentPadding: const EdgeInsets.all(0.0),
+                          title: Text(
+                            S.current.enablePINLogin,
+                            style: TextStyle(color: mainTextColor),
+                          ),
+                          subtitle: Text(
+                            S.current.pinLoginSetupInstructions,
+                            style: TextStyle(color: mainTextColor),
+                          ),
+                          value: isPINLoginEnabledValue ?? false,
+                          onChanged: (value) async {
+                            if (userId == null) {
+                              showToast(S.current.unexpectedErrorOccured);
+                              return;
+                            }
+
+                            if (userId == GuestUserDetails.guestUserId) {
+                              showToast(S.current
+                                  .pleaseSetupYourAccountToUseThisFeature);
+                              return;
+                            }
+                            if (value) {
+                              // Call the PIN reset popup
+                              await pinResetPopup(
+                                context: context,
+                                userPinId: userId,
+                              );
+                            }
+
+                            try {
+                              userConfigCubit.setUserConfig(
+                                UserConfigConstants.isPINLoginEnabled, //change
+                                value,
+                              );
```

Comment:
Currently the switch gets enabled even if pin is not successfully set, you can return a boolean value from `pinResetPopup` using `Navigator.pop(context)` and enable this switch only if its true. You can use `passwordLoginPopup` as reference for returning data from `Navigator.pop(context)`